### PR TITLE
Handle coverage the same was as exunit

### PR DESCRIPTION
**Pull request by toland**
This allows the end user to use custom coverage tools.

### CODE REVIEW

File: lib/mix/tasks/eunit.ex [hunk 1]
```diff
@@ -36,25 +36,36 @@ defmodule Mix.Tasks.Eunit do
...
+
     # run the actual tests
-    if(options[:cover], do: cover_start())
     test_modules(post_config[:erlc_paths], options[:patterns])
     |> Enum.map(&module_name_from_path/1)
     |> Enum.drop_while(fn(m) ->
       tests_pass?(m, options[:eunit_opts] ++ post_config[:eunit_opts]) end)
-    if(options[:cover], do: cover_analyse())
+
+    cover && cover.()
```
Comment (dantswain) [inline lib/mix/tasks/eunit.ex]: I don't get how this works.  It looks like cover is either `nil` or a Keyword.  How would `cover.()` work?

Comment (toland) [reply lib/mix/tasks/eunit.ex]: `cover` is either `nil` or a function. `Cover.start/2` returns a function that does the analysis and writes the output.

The code for the default `Cover` module is here: https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/tasks/test.ex#L2

Comment (dantswain) [reply lib/mix/tasks/eunit.ex]: OK, so I think this is just a bit of funkyness with variable names (like having a couple things named `cover`).  Can we do something like

``` elixir
def run(args) do
   # .. existing code

  # line 55
  cover_callback = start_cover_tool(options[:cover], project)

  # lines 63-66 same

  # line 68
  execute_cover_callback(cover_callback)
end

# no cover tool specified
defp start_cover_tool(nil, _project), do: nil
# set up the cover tool
defp start_cover_tool(cover_tool, project) do
  compile_path = Mix.Project.compile_path(project)
  cover = Keyword.merge(@cover, project[:test_coverage] || [])
  # returns a callback
  cover[:tool].start(compile_path, cover)
end

# no cover tool was specified
defp execute_cover_callback(nil), do: :ok
# run the cover callback
defp execute_cover_callback(cb), do: cb.()
```

**Comment by toland** [comment]
@dantswain Done. Let me know what you think.

**Comment by dantswain** [comment]
:+1: :shipit: and Thanks!

**Pull request merged**